### PR TITLE
bug: add handling for back button to close in-app message

### PR DESCRIPTION
**Pull request by vincentdelmundo**
# Description
* add key event listener for back button in base view
* add removal of message and update of repository on tap of back button

## Links
https://jira.rakuten-it.com/jira/browse/SDKCF-2308

# Checklist
- [x] I have read the [contributing guidelines](../CONTRIBUTING.md).
- [x] I wrote/updated tests for new/changed code
- [x] I ran `./gradlew check` without errors

### CODE REVIEW

File: inappmessaging/src/test/java/com/rakuten/tech/mobile/inappmessaging/runtime/view/InAppMessageViewListenerSpec.kt [hunk 1]
```diff
@@ -193,4 +196,88 @@ class InAppMessageViewListenerSpec : BaseTest() {
 
         listener.onTouch(mockView, mockEvent).shouldBeFalse()
     }
+
+    @Test
```
Comment (corycaywood) [inline inappmessaging/src/test/java/com/rakuten/tech/mobile/inappmessaging/runtime/view/InAppMessageViewListenerSpec.kt:200]: This test class is getting pretty long. Sometimes in this case I will split up the tests to have multiple classes within the same file. It helps the tests be more organized and easier to read. i.e. these new tests could be in a class named `InAppMessageViewListenerBackButtonSpec`. 

See here for an example of what I'm talking about: https://github.com/rakutentech/android-remote-config/blob/master/remote-config/src/test/kotlin/com/rakuten/tech/mobile/remoteconfig/RealRemoteConfigSpec.kt

File: inappmessaging/src/test/java/com/rakuten/tech/mobile/inappmessaging/runtime/view/InAppMessageViewListenerSpec.kt [hunk 2]
```diff
@@ -193,4 +196,88 @@ class InAppMessageViewListenerSpec : BaseTest() {
...
+        fun `should return true with back button click`() {
+        val mockDisplayManager = Mockito.mock(DisplayManager::class.java)
```
Comment (corycaywood) [inline inappmessaging/src/test/java/com/rakuten/tech/mobile/inappmessaging/runtime/view/InAppMessageViewListenerSpec.kt:202]: It looks like this mock setup could be moved either to a utility function or setup in the `@Before` block to reduce duplication.

File: inappmessaging/src/test/java/com/rakuten/tech/mobile/inappmessaging/runtime/view/InAppMessageViewListenerSpec.kt (hunk 1 above)
Comment (vincentdelmundo) [reply inappmessaging/src/test/java/com/rakuten/tech/mobile/inappmessaging/runtime/view/InAppMessageViewListenerSpec.kt]: Created separate class for OnKey and OnTouch

File: inappmessaging/src/test/java/com/rakuten/tech/mobile/inappmessaging/runtime/view/InAppMessageViewListenerSpec.kt (hunk 2 above)
Comment (vincentdelmundo) [reply inappmessaging/src/test/java/com/rakuten/tech/mobile/inappmessaging/runtime/view/InAppMessageViewListenerSpec.kt]: Moved to class variable

**Pull request merged**